src/hooks/interfaces.ts

```diff
@@ -0,0 +1,31 @@
+import { IFormatOptionQuery } from '@/utils/getPagiantion';
+import { UseQueryOptions } from 'react-query';
+
+export interface PaginationServicer<T> {
+  getPagination: (query: IFormatOptionQuery) => Promise<T>;
+}
+
+export interface Pagination<T> {
+  data: T[];
+  meta: Meta;
+}
+
+export interface Meta {
+  page: number;
+  take: number;
+  itemCount: number;
+  pageCount: number;
+  hasPreviousPage: boolean;
+  hasNextPage: boolean;
+}
+
+export interface IBaseUseReactQuery<T> {
+  dependent: [any];
```

Comment:
We should use noun `dependencies` or name `queryKeys` instead of `dependent`

Reply:
thanks